docs/content/preview/tutorials/AI/ai-llamaindex-openai.md

```diff
@@ -0,0 +1,250 @@
+    -p 15435:15433 -p 7003:7000 -p 9003:9000 -p 5435:5433 \
+    -v ~/yb_docker_data/node3:/home/yugabyte/yb_data --restart unless-stopped \
+    yugabytedb/yugabyte:2.20.1.0-b97 \
+    bin/yugabyted start --join=yugabytedb-node1 \
+    --base_dir=/home/yugabyte/yb_data --background=false
+```
+
+The database connectivity settings are provided in the `{project_dir}/.env` file and do not need to be changed if you started the cluster with the preceding command.
+
+Navigate to the YugabyteDB UI to confirm that the database is up and running, at <http://127.0.0.1:15433>.
+
+## Load the Financial Schema and Seed Data
+
+This application requires a database table with financial information for companies in the S&P 500. This schema includes a `companies` table. It also creates a read-only user role to prevent any destructive actions while querying the database directly from LlamaIndex.
+
+1. Copy the schema to the first node's Docker container.
+    ```sh
+    docker cp {project_dir}/sql/schema_simplified.sql yugabytedb-node1:/home
+    ```   
+
+2. Copy the seed data file to the Docker container.
+    ```sh
+    docker cp {project_dir}/sql/data_simplified.sql yugabytedb-node1:/home
+    ```
+
+3. Execute the SQL files against the database.
+    ```sh
+     docker exec -it yugabytedb-node1 bin/ysqlsh -h yugabytedb-node1 -c '\i /home/schema_simplified.sql'
+     docker exec -it yugabytedb-node1 bin/ysqlsh -h yugabytedb-node1 -c '\i /home/data_simplified.sql'
+    ```
+
+## Start the Application
+
+This command-line application takes an input in natural language and returns a response from LlamaIndex.
+
+1. Start the server.
+
+```sh
+python3 index.py
+```
+
+```output
+What is your question?
+```
+
+2. Provide a relevant question. For instance:
+
+```
+What is your question? 
+
+Provide a detailed company history for the company with the highest marketcap.
+```
+
+```output
+Querying SQL database: The first choice seems more relevant as it mentions translating a natural language query into a SQL query over a table containing companies' stats. This could potentially include the company with the highest marketcap and provide a detailed history. The second choice is more about answering semantic questions, which doesn't necessarily imply detailed company history.
+
+SQL query: SELECT * 
+FROM companies_simplified 
+WHERE marketcap = (SELECT MAX(marketcap) FROM companies_simplified)
+
+SQL response: The company with the highest marketcap is Microsoft Corporation. It was founded on April 4, 1975, and is headquartered at One Microsoft Way, Redmond, Washington, United States. Microsoft is a technology company that specializes in software infrastructure. It has a marketcap of $1,043,526,401,920 and employs 221,000 people. The company's contact number is 425-882-8080.
+
+Transformed query given SQL response: Can you provide more details about the key products and services offered by Microsoft Corporation?
```

Comment:
Btw, is this a follow-up question from you or the thought-process of the AI agent? Not clear.

Reply:
All of this `output` is generated by the AI agent. A new question is posed after first using SQL to determine that Microsoft is the company with the highest market cap.